### PR TITLE
P20-1273: Track user level progress with current locale

### PR DESCRIPTION
**Pull request by artem-vavilov**
The RED team often wants to know which language a user selects after completing a level.
Currently, we only have the locale field in the users table, which records the user's first language selection site-wide. However, we need to track language preferences at the user-level granularity as different curricula are translated.

## Links
- JIRA task: [P20-1273](https://codedotorg.atlassian.net/browse/P20-1273)
- Closed PR: https://github.com/code-dot-org/code-dot-org/pull/62310

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -2133,6 +2144,11 @@ def self.track_level_progress(user_id:, level_id:, script_id:, new_result:, subm
       total_time_spent = user_level.calculate_total_time_spent(time_spent)
       user_level.time_spent = total_time_spent if total_time_spent
 
+      if locale
+        user_level.locale = locale
+        user_level.locale_supported = script.supported_locale?(locale)
```
Comment (artem-vavilov) [inline dashboard/app/models/user.rb:2149]: @code-dot-org/red-team Please let me know if you have a better name for the flag (e.g. `supports_locale`)

File: dashboard/app/models/unit.rb [hunk 2]
```diff
@@ -1985,6 +1985,10 @@ def supported_locale_codes
     locales.sort
   end
 
+  def supported_locale?(locale)
+    supported_locale_codes.include?(locale.to_s)
```
Comment (artem-vavilov) [inline dashboard/app/models/unit.rb:1989]: https://github.com/code-dot-org/code-dot-org/blob/085fca331b9a94051a78d876ab01c192aadc5e66/dashboard/app/models/unit.rb#L1982-L1986

Comment (artem-vavilov) [inline dashboard/app/models/unit.rb:1989]: https://github.com/code-dot-org/code-dot-org/blob/085fca331b9a94051a78d876ab01c192aadc5e66/dashboard/app/models/unit.rb#L1982-L1986

File: dashboard/app/models/user.rb (hunk 1 above)
Comment (carl-codeorg) [reply dashboard/app/models/user.rb:2149]: I think `locale_supported` is actually pretty good. The flag denotes whether or not the language was supported at the time the user attempted the level, right? So the past tense makes more sense to me.

**Review by carl-codeorg — APPROVED** [review]

**Pull request merged**